tools/integration-test/src/tests/wasm_client.rs

```diff
@@ -0,0 +1,210 @@
+
+#[test]
+fn test_create_and_update_wasm_client() -> Result<(), Error> {
+    run_binary_node_test(&CreateAndUpdateWasmClientTest)
+}
+
+const WASM_PATH: &str = "fixtures/wasm/ibc_client_tendermint_cw.wasm";
+
+const TM_CREATE_OPTIONS: TendermintCreateOptions = TendermintCreateOptions {
+    max_clock_drift: Some(Duration::from_secs(3)),
+    trusting_period: Some(Duration::from_secs(60)),
+    trust_threshold: Some(TrustThreshold::TWO_THIRDS),
+};
+
+fn sha256(data: &[u8]) -> Vec<u8> {
+    use sha2::Digest;
+    let mut hasher = sha2::Sha256::new();
+    hasher.update(data);
+    hasher.finalize().to_vec()
+}
+
+struct CreateAndUpdateWasmClientTest;
+
+impl TestOverrides for CreateAndUpdateWasmClientTest {
+    fn client_options_a_to_b(&self) -> CreateOptions {
+        let wasm_code = std::fs::read(WASM_PATH).unwrap();
+        let checksum = sha256(&wasm_code);
+
+        WasmCreateOptions {
+            checksum,
+            underlying: TM_CREATE_OPTIONS.into(),
+        }
+        .into()
+    }
+
+    fn client_options_b_to_a(&self) -> CreateOptions {
+        let wasm_code = std::fs::read(WASM_PATH).unwrap();
+        let checksum = sha256(&wasm_code);
+
+        WasmCreateOptions {
+            checksum,
+            underlying: TM_CREATE_OPTIONS.into(),
+        }
+        .into()
+    }
+
+    fn should_spawn_supervisor(&self) -> bool {
+        false
+    }
+
+    fn modify_genesis_file(&self, genesis: &mut serde_json::Value) -> Result<(), Error> {
+        set_max_deposit_period(genesis, "10s")?;
+        set_voting_period(genesis, 10)?;
+        add_allowed_client(genesis, "08-wasm")?;
+
+        Ok(())
+    }
+
+    fn modify_node_config(&self, config: &mut toml::Value) -> Result<(), Error> {
+        set_max_body_bytes(config, 10001048576)?;
+
+        Ok(())
+    }
+}
+
+impl BinaryNodeTest for CreateAndUpdateWasmClientTest {
+    fn run(
+        &self,
+        test_config: &TestConfig,
+        node_a: FullNode,
+        node_b: FullNode,
+    ) -> Result<(), Error> {
+        let (chain_a, chain_b) = setup_chains(test_config, &node_a, &node_b)?;
+
+        let _drop_handle_a = DropChainHandle(chain_a.clone());
+        let _drop_handle_b = DropChainHandle(chain_b.clone());
+
+        info!("Storing Wasm contract on chain A");
+        store_wasm_contract(node_a, test_config)?;
+
+        info!("Storing Wasm contract on chain B");
+        store_wasm_contract(node_b, test_config)?;
+
+        let overrides = self.get_overrides();
+
+        info!("Creating client A to B");
+        let mut client_a_to_b =
+            bootstrap_foreign_client(&chain_a, &chain_b, overrides.client_options_a_to_b())?;
+
+        info!("Creating client B to A");
+        let mut client_b_to_a =
+            bootstrap_foreign_client(&chain_b, &chain_a, overrides.client_options_b_to_a())?;
+
+        info!("Refreshing client A to B...");
+        let res = client_a_to_b.refresh(true);
```

Comment:
Could we use `client_refresh_rate` and wait for the appropriate amount of time https://github.com/informalsystems/hermes/blob/v1.8.2/config.toml#L356?
Related to the comment under `try_refresh()`

Reply:
See my comment above.